sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/PubsubHelper.java

```diff
@@ -0,0 +1,231 @@
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubOptions;
+import org.apache.beam.sdk.util.BackOff;
+import org.apache.beam.sdk.util.BackOffUtils;
+import org.apache.beam.sdk.util.FluentBackoff;
+import org.apache.beam.sdk.util.Sleeper;
+import org.joda.time.Duration;
+
+/**
+ * Helper for working with pubsub and gcs.
+ */
+public class PubsubHelper {
+  /** Underlying pub/sub client. */
+  private final PubsubClient pubsubClient;
+
+  /** Project id. */
+  private final String project;
+
+  /** Topics we should delete on close. */
+  private final List<TopicPath> createdTopics;
+
+  /** Subscriptions we should delete on close. */
+  private final List<SubscriptionPath> createdSubscriptions;
+
+  /** How to sleep in retry loops. */
+  private final Sleeper sleeper;
+
+  /** How to backoff in retry loops. */
+  private final BackOff backOff;
+
+  private PubsubHelper(PubsubClient pubsubClient, String project) {
+    this.pubsubClient = pubsubClient;
+    this.project = project;
+    createdTopics = new ArrayList<>();
+    createdSubscriptions = new ArrayList<>();
+    sleeper = Sleeper.DEFAULT;
+    backOff = FluentBackoff.DEFAULT
+      .withInitialBackoff(Duration.standardSeconds(1))
+      .withMaxRetries(3)
+      .backoff();
+  }
+
+  /** Create a helper. */
+  public static PubsubHelper create(PubsubOptions options) throws IOException {
+    return new PubsubHelper(
+      PubsubJsonClient.FACTORY.newClient(null, null, options),
+      options.getProject());
+  }
+
+  /**
+   * Create a topic from short name. Delete it if it already exists. Ensure the topic will be
+   * deleted on cleanup. Return full topic name.
+   * @throws InterruptedException
+   */
+  public TopicPath createTopic(String shortTopic) throws IOException {
+    TopicPath topic = PubsubClient.topicPathFromName(project, shortTopic);
+    while (true) {
+      try {
+        NexmarkUtils.console("create topic %s", topic);
+        pubsubClient.createTopic(topic);
+        createdTopics.add(topic);
+        return topic;
+      } catch (GoogleJsonResponseException ex) {
+        NexmarkUtils.console("attempting to cleanup topic %s", topic);
+        pubsubClient.deleteTopic(topic);
+        try {
+          if (!BackOffUtils.next(sleeper, backOff)) {
+            NexmarkUtils.console("too many retries for creating topic %s", topic);
+            throw ex;
+          }
+        } catch (InterruptedException in) {
+          throw new IOException(in);
+        }
+      }
+    }
+  }
+
+  /** Create a topic from short name if it does not already exist. The topic will not be
+   * deleted on cleanup. Return full topic name.
+   * @throws InterruptedException */
+  public TopicPath createOrReuseTopic(String shortTopic) throws IOException {
+    TopicPath topic = PubsubClient.topicPathFromName(project, shortTopic);
+    while (true) {
+      try {
+        NexmarkUtils.console("create topic %s", topic);
+        pubsubClient.deleteTopic(topic);
```

Comment:
Do you mean createTopic ?

Reply:
Good catch. I didn't test code with existing topics.